modules/registration/registration_api.rb

```diff
@@ -0,0 +1,34 @@
+require 'registration/proxy_request'
+
+class Proxy::Registration::Api < ::Sinatra::Base
+  get '/' do
+    response = Proxy::Registration::ProxyRequest.new.global_register(request)
+    handle_response(response)
+  rescue StandardError => e
+    logger.error "Error when rendering Global Registration Template: #{e}"
+    render_error
+  end
+
+  post '/' do
+    response = Proxy::Registration::ProxyRequest.new.host_register(request)
+    handle_response(response)
+  rescue StandardError => e
+    logger.error "Error when rendering Host Registration Template: #{e}"
+    render_error
+  end
+
+  private
+
+  def handle_response(response)
+    if response.code.start_with? '2'
+      response.body
+    else
+      render_error(response.code, response.body)
```

Comment:
Would it be useful log something here?

Another thing that came to mind is look at the content type of the response. If it's `text/html`, you don't want to pass the body but if it's `text/plain` it's probably safe to pass along. This may help when Apache generates some HTTP 502 error (in HTML) vs something that the Foreman application did. It may require a change on the Foreman side as well.